### DISCUSSION TITLE
Proxmox Backup Server 3

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#1561] [Announcements]
This command is designed to transition a currently running PBS 2 instance into a running PBS 3 instance.
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/pbs3-upgrade.sh)"
```
To ensure a proper source installation, please respond with "yes" to all questions prompted during the installation process.

**Comment by eteixido1** [comment]
Hello tteck,

could you please let me know if could be possible to have a PBS install script on LXC?

Kind regards,

**Reply by tteck** [reply]
It is not recommended to store backups on the same disk where Proxmox is running, as it is typically local by default. It is advisable to save backups to a USB SSD drive. It is worth noting that this solution does not require PBS, which I have personally never needed or used. So, my answer would be no.

Worth a read: https://smarthomescene.com/guides/how-to-backup-home-assistant-in-proxmox/

**Reply by tteck** [reply]
My instances are backed up every 4 hours to a USB SSD drive with a retention of 24 backups. As a result, I have 4 days' worth of backups at 4-hour intervals. 10 LXC containers using less than 250GB.

**Reply by eteixido1** [reply]
Thank you tteck, I'll use the guide to configure my backup policies. It will be enough for me.

**Reply by timdonovanuk** [reply]
But it should be possible to run PBS in an LXC right? This person didn't neccessarily say they were going to store the backup data on the same disk as the host. I imagine many people use network storage with PBS (I use a remote iscsi connection to my NAS).

**Comment by hubsi-k** [comment]
Are there any possible problems with the script?

Hit:1 http://security.debian.org/debian-security bookworm-security InRelease
Get:2 http://download.proxmox.com/debian/pbs bookworm InRelease [2,782 B]
Hit:3 http://deb.debian.org/debian bookworm InRelease
Hit:4 http://deb.debian.org/debian bookworm-updates InRelease
Err:2 http://download.proxmox.com/debian/pbs bookworm InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 1140AF8F639E0C39
Reading package lists... Done
W: GPG error: http://download.proxmox.com/debian/pbs bookworm InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 1140AF8F639E0C39
E: The repository 'http://download.proxmox.com/debian/pbs bookworm InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
root@pbs:~#

**Reply by SlothCroissant** [reply]
FYI, this worked for me today without issue. Maybe there was an issue with the Proxmox repositories a couple days back? 

Either way, likely not an issue with the script, since it hasn't had commits since last week at this point.

**Reply by hubsi-k** [reply]
Thx, for feedback - it has now worked for me too.
